skylines/frontend/views/flight.py

```diff
@@ -448,26 +448,39 @@ def update():
     if 'pilot_id' in data:
         pilot_id = data['pilot_id']
 
-        if pilot_id is not None and not User.exists(id=pilot_id):
-            return jsonify(error='unknown-pilot'), 422
+        if pilot_id is not None:
 
-        if g.flight.pilot_id != pilot_id:
-            g.flight.pilot_id = pilot_id
+            if not User.exists(id=pilot_id):
+                return jsonify(error='unknown-pilot'), 422
 
-            # update club if pilot changed
-            if pilot_id is not None:
+            if g.flight.pilot_id != pilot_id and User.get(pilot_id).club_id == g.current_user.club_id:
```

Comment:
I'm not sure about this change. I think it would be better to return an error if the clubs don't match. e.g. `return jsonify(error='pilot-disallowed'), 422`